internal/sql/pg/query.go

```diff
@@ -200,7 +195,11 @@ func query(ctx context.Context, cq connQueryer, stmt string, args ...any) (*sql.
 		if err != nil {
 			return nil, err
 		}
-		return decodeFromPGType(pgxVals...)
+		oids := make([]uint32, len(pgxVals))
+		for i, v := range row.FieldDescriptions() {
+			oids[i] = v.DataTypeOID
+		}
+		return decodeFromPG(pgxVals, oids, oidToDataType)
```

Comment:
In most cases, I bet this approach is the simplest and least error prone.  We might be able to cut out the middleman somewhat and lift the implementation of `Values` from pgx/rows.go, but to make that much more efficient than this, we'd be in the weeds writing codecs that we'd have to maintain.

Reply:
What you're doing.   Compared to trying to decode the `RawValues` ourselves.  Just going through pgtypes feels best despite alternatives existing that might avoid conversion.